gymnasium/core.py

```diff
@@ -279,31 +281,46 @@ def __getattr__(self, name: str) -> Any:
     @property
     def spec(self) -> EnvSpec | None:
         """Returns the :attr:`Env` :attr:`spec` attribute with the `WrapperSpec` if the wrapper inherits from `EzPickle`."""
-        env_spec = self.env.spec
-
-        if env_spec is not None:
-            from gymnasium.envs.registration import WrapperSpec
-
-            # See if the wrapper inherits from `RecordConstructorArgs` then add the kwargs otherwise use `None` for the wrapper kwargs. This will raise an error in `make`
-            if isinstance(self, RecordConstructorArgs):
-                kwargs = getattr(self, "_saved_kwargs")
-                if "env" in kwargs:
-                    kwargs = deepcopy(kwargs)
-                    kwargs.pop("env")
-            else:
-                kwargs = None
-
-            wrapper_spec = WrapperSpec(
-                name=self.class_name(),
-                entry_point=f"{self.__module__}:{type(self).__name__}",
-                kwargs=kwargs,
-            )
-
-            # to avoid reference issues we deepcopy the prior environments spec and add the new information
-            env_spec = deepcopy(env_spec)
-            env_spec.applied_wrappers += (wrapper_spec,)
+        if self._cached_spec is None:
```

Comment:
Minor comment for slight unspagghettification - it might be better to do
```python
if self._cached_spec is not None:
    return self._cached_spec

env_spec = self.env.spec
...
```

Basically just flip the condition and do an early return to remove one level of indentation from the whole code block.

Reply:
Updated